### PR TITLE
scripts: update demo records loading

### PR DESCRIPTION
**Pull request by hachreak**
Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: requirements-devel.txt [hunk 1]
```diff
@@ -26,7 +26,7 @@
 -e git+https://github.com/inveniosoftware/invenio-access.git#egg=invenio-access
 -e git+https://github.com/inveniosoftware/invenio-accounts.git#egg=invenio-accounts
 -e git+https://github.com/inveniosoftware/invenio-assets.git#egg=invenio-assets
--e git+https://github.com/inveniosoftware/invenio-base.git#egg=invenio-base
+-e git+https://github.com/hachreak/invenio-base.git@add_search_index#egg=invenio-base
```
Comment (lnielsen) [inline requirements-devel.txt]: I guess it shouldn't be hackreak here :-)

Comment (hachreak) [inline requirements-devel.txt]: yes, should not.
I wait that these are merged.

File: requirements-devel.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@
...
 -e git+https://github.com/inveniosoftware/invenio-celery.git#egg=invenio-celery
```
Comment (tiborsimko) [inline requirements-devel.txt]: Note: this is WIP untitl dependent PRs are merged.

**Comment by hachreak** [comment]
Waiting for:
 - [ ] templates: update record detail https://github.com/inveniosoftware/invenio-records-ui/pull/29
 - [ ] cli: reindex function addition https://github.com/inveniosoftware/invenio-indexer/pull/9
 - [ ] demo: search engine packages installation https://github.com/inveniosoftware/invenio-base/pull/58

**Comment by hachreak** [comment]
ping @jirikuncar :shipit:

File: requirements-devel.txt [hunk 3]
```diff
@@ -41,3 +41,6 @@
 -e git+https://github.com/inveniosoftware/invenio-rest.git#egg=invenio-rest
 -e git+https://github.com/inveniosoftware/invenio-search.git#egg=invenio-search
 -e git+https://github.com/inveniosoftware/invenio-theme.git#egg=invenio-theme
+-e git+https://github.com/inveniosoftware/invenio-indexer.git#egg=invenio-indexer
```
Comment (jirikuncar) [inline requirements-devel.txt]: can you keep it sorted?

File: scripts/create-instance.sh [hunk 4]
```diff
@@ -108,12 +108,14 @@ set -o nounset
 
 if [[ "$@" != *"--devel"* ]]; then
 # sphinxdoc-install-invenio-full-begin
+
```
Comment (jirikuncar) [inline scripts/create-instance.sh]: Please remove the empty line as it doesn't look nice in generated docs. Thanks

File: scripts/create-instance.sh [hunk 5]
```diff
@@ -147,6 +149,49 @@ echo "CELERY_ACCEPT_CONTENT=['json', 'msgpack', 'yaml']" >> ../../var/${INVENIO_
...
+RECORDS_REST_ENDPOINTS = dict(
+    recid=dict(
+        pid_type='recid',
+        pid_minter='recid',
+        pid_fetcher='recid',
+        search_index='marc21',
+        search_type=None,
+        record_serializers={
+        'application/json': ('invenio_records_rest.serializers'
+            ':json_v1_response'),
```
Comment (jirikuncar) [inline scripts/create-instance.sh]: can you verify the indentation?

File: scripts/create-instance.sh [hunk 6]
```diff
@@ -147,6 +149,49 @@ echo "CELERY_ACCEPT_CONTENT=['json', 'msgpack', 'yaml']" >> ../../var/${INVENIO_
...
+        },
+        search_serializers={
+        'application/json': ('invenio_records_rest.serializers'
+            ':json_v1_search'),
```
Comment (jirikuncar) [inline scripts/create-instance.sh]: can you verify the indentation?

File: scripts/populate-instance.sh [hunk 7]
```diff
@@ -113,22 +113,24 @@ ${INVENIO_WEB_INSTANCE} users create \
...
 # sphinxdoc-populate-with-demo-records-end
 
-# sphinxdoc-register-pid-begin
-recid=1
-for demouuid in $demouuids; do
-    ${INVENIO_WEB_INSTANCE} pid create \
-         -t rec -i $demouuid -s REGISTERED recid $recid
-    let recid=recid+1
-done
-# sphinxdoc-register-pid-end
```
Comment (jirikuncar) [inline scripts/populate-instance.sh]: please find the section `# sphinxdoc-register-pid-begin` in documentation and update the text around mentioning `--pid-minter recid`

Comment (hachreak) [inline scripts/populate-instance.sh]: done

File: docs/installation/installation-detailed.rst [hunk 8]
```diff
@@ -598,12 +598,11 @@ taken from the ``invenio-records`` package:
    :end-before: # sphinxdoc-populate-with-demo-records-end
    :literal:
 
-Let's register incremental numerical persistent identifiers for the uploaded
-demo records:
+Let's index all the records uploaded:
```
Comment (jirikuncar) [inline docs/installation/installation-detailed.rst]: What about: `Let's index all uploaded record:`

File: scripts/create-instance.sh [hunk 9]
```diff
@@ -114,6 +114,7 @@ pip install invenio-access[postgresql] --pre
 pip install invenio-search --pre
 pip install dojson --pre
 # now we can install full Invenio:
+
```
Comment (jirikuncar) [inline scripts/create-instance.sh]: you can remove this empty line

File: scripts/populate-instance.sh (hunk 7 above)
Comment (jirikuncar) [inline scripts/populate-instance.sh]: Thanks

File: scripts/populate-instance.sh [hunk 10]
```diff
@@ -117,18 +117,18 @@ demomarc21pathname=$(echo "from __future__ import print_function; \
...
+# sphinxdoc-index-all-records-begin
+# indexing
+invenio3 index init
```
Comment (jirikuncar) [inline scripts/populate-instance.sh]: `s/invenio3/${INVENIO_WEB_INSTANCE}/`

**Comment by hachreak** [comment]
ping @jirikuncar

**Comment by tiborsimko** [comment]
@hachreak I tried this PR with docker-machine and run into issues (e.g. no search results). Let's see together IRL...

**Comment by tiborsimko** [comment]
Can you please add new static folder to the `docker-compose.yml`?  

```
   volumes:
     - /home/invenio/.virtualenvs/invenio3/var/invenio3-instance/static
     - /home/invenio/.virtualenvs/invenio3/src/invenio-theme/invenio_theme/static
+    - /home/invenio/.virtualenvs/invenio3/src/invenio-search-ui/invenio_search_ui/static
```

File: scripts/populate-instance.sh [hunk 11]
```diff
@@ -117,18 +117,18 @@ demomarc21pathname=$(echo "from __future__ import print_function; \
...
-for demouuid in $demouuids; do
-    ${INVENIO_WEB_INSTANCE} pid create \
-         -t rec -i $demouuid -s REGISTERED recid $recid
-    let recid=recid+1
-done
-# sphinxdoc-register-pid-end
+# sphinxdoc-index-all-records-begin
+# indexing
+${INVENIO_WEB_INSTANCE} index init
+sleep 10
```
Comment (tiborsimko) [inline scripts/populate-instance.sh]: FWIW this did not seem enough for a remote container on CERN OpenStack VM.  Perhaps increase the timeout to 20 seconds?

Comment (tiborsimko) [inline scripts/populate-instance.sh]: Hmm, not helping... I had to manually launch:

```console
$ docker exec -i -t invenio_web_1 /bin/bash
(invenio3) invenio@a2b2f18dfb49:/code$ invenio3 index reindex --yes-i-know  
Sending records to indexing queue ...
Execute "run" command to process the queue!
(invenio3) invenio@a2b2f18dfb49:/code$ invenio3 index run  
```

**Comment by tiborsimko** [comment]
Another thing is the page header:

Here is the home page before starting search:

2016-03-18-155606_2560x1440_scrot

Here is the search results page: (note bad header)

2016-03-18-155655_2560x1440_scrot

Here is the detailed record page: (the header is good again)

2016-03-18-155740_2560x1440_scrot

(BTW note also leading comma before author name...)

File: scripts/populate-instance.sh (hunk 11 above)
Comment (tiborsimko) [inline scripts/populate-instance.sh]: @hachreak If you don't find the problem (e.g. check all configuration variables?) then you can perhaps remove those sleep times and add these reindexing instructions manually after the population step... in this way people know what they have to do until we properly fix things.

**Comment by hachreak** [comment]
@tiborsimko for now I'll run 2 times the indexer on `./scripts/populate-instance.sh`.
This "resolve" the problem.
The question now is: "why the theme disappear on search?"

**Comment by tiborsimko** [comment]
> @tiborsimko for now I'll run 2 times the indexer on ./scripts/populate-instance.sh.
> This "resolve" the problem.

Perfect, that will do for now.

> The question now is: "why the theme disappear on search?"

Maybe check latest commits in concerned repositories? Later we could also switch to using PyPI-released versions in the demo site, with the git master versions the things are expected to be a bit fragile...

**Comment by hachreak** [comment]
I temporary fixed the indexing issue (and opened a [issue](https://github.com/inveniosoftware/invenio-indexer/issues/11) on `invenio-indexer` to find a solution), fixed invenio-base cookiecutter template and now works.
@tiborsimko when you have time, can you test it on your server? thanks

File: scripts/populate-instance.sh [hunk 12]
```diff
@@ -117,18 +117,19 @@ demomarc21pathname=$(echo "from __future__ import print_function; \
...
-    ${INVENIO_WEB_INSTANCE} pid create \
-         -t rec -i $demouuid -s REGISTERED recid $recid
-    let recid=recid+1
-done
-# sphinxdoc-register-pid-end
+# sphinxdoc-index-all-records-begin
+# indexing
+${INVENIO_WEB_INSTANCE} index init
+sleep 20
+${INVENIO_WEB_INSTANCE} index run
```
Comment (jirikuncar) [inline scripts/populate-instance.sh]: please use `${INVENIO_WEB_INSTANCE} index queue init` with Invenio-Indexer 1.0.0a2 or latest master

File: scripts/populate-instance.sh [hunk 13]
```diff
@@ -117,18 +117,19 @@ demomarc21pathname=$(echo "from __future__ import print_function; \
...
-    ${INVENIO_WEB_INSTANCE} pid create \
-         -t rec -i $demouuid -s REGISTERED recid $recid
-    let recid=recid+1
-done
-# sphinxdoc-register-pid-end
+# sphinxdoc-index-all-records-begin
+# indexing
+${INVENIO_WEB_INSTANCE} index init
+sleep 20
+${INVENIO_WEB_INSTANCE} queue init
```
Comment (jirikuncar) [inline scripts/populate-instance.sh]: `index queue init`

**Comment by hachreak** [comment]
@tiborsimko issues resolved completely.

**Pull request merged**